packages/cli/src/lib/implementation/config-middleware.spec.ts

```diff
@@ -52,3 +58,110 @@ describe('applyConfigMiddleware', () => {
     expect(error?.message).toContain(defaultConfigPath);
   });
 });
+
+describe('filterPluginsByOnlyPluginsOption', () => {
+  it('should return all plugins if no onlyPlugins option', async () => {
+    const plugins = [
+      { slug: 'plugin1' },
+      { slug: 'plugin2' },
+      { slug: 'plugin3' },
+    ];
+    const filtered = filterPluginsByOnlyPluginsOption(
+      plugins as CoreConfig['plugins'],
+      {},
+    );
+    expect(filtered).toEqual(plugins);
+  });
+
+  it('should return only plugins with matching slugs', () => {
+    const plugins = [
+      { slug: 'plugin1' },
+      { slug: 'plugin2' },
+      { slug: 'plugin3' },
+    ];
+    const filtered = filterPluginsByOnlyPluginsOption(
+      plugins as CoreConfig['plugins'],
+      {
+        onlyPlugins: ['plugin1', 'plugin3'],
+      },
+    );
+    expect(filtered).toEqual([{ slug: 'plugin1' }, { slug: 'plugin3' }]);
+  });
+});
+
+// without the `no-secrets` rule, this would be flagged as a security issue
+// eslint-disable-next-line no-secrets/no-secrets
+describe('filterCategoryByOnlyPluginsOption', () => {
+  let logSpy: SpyInstance;
+  beforeEach(() => {
+    logSpy = vi.spyOn(console, 'log');
+  });
+
+  afterEach(() => {
+    logSpy.mockClear();
```

Comment:
```suggestion
    logSpy.mockRestore();
```

Otherwise it won't revert to the original implementation.